File: models/member.py
from datetime import date
from app import db

class MemberModel(db.Model):
	__tablename__ = 'Member'

	id = db.Column(db.Integer, primary_key=True)

	email = db.Column(db.String)
	first_name = db.Column(db.String, nullable=False)
	last_name = db.Column(db.String, nullable=False)
	gender = db.Column(db.String)

	joined_on = db.Column(db.Date, default=date.today) #passing function so the value of `today` is the time when a new row is added.

	access = db.Column(db.Integer, unique=True) # TODO: Validate access numbers
	sid = db.Column(db.Integer, unique=True)

	newsletter = db.Column(db.Boolean, default=True)
	doing_it = db.Column(db.Boolean, default=False)
	registered = db.Column(db.Boolean, default=False)

	events_attended = db.relationship('AttendanceModel', back_populates='member')

	@db.validates('email')
	def validate_email(self, key, email):
		assert '@' in email
		return email

	@db.validates('sid')
	def validate_sid(self, key, sid):
		str_sid = str(sid)
		assert len(str_sid) == 9
		return sid

File: src/entities/member.ts
import { Entity, Column, PrimaryGeneratedColumn, OneToMany } from 'typeorm';
import { AttendanceEntity } from './attendance'
import { Gender } from 'interfaces/member/gender';
import { Member } from 'interfaces/member';
import { BaseEntity } from './entity.base';

@Entity({name: "Member"})
export class MemberEntity extends BaseEntity implements Member {

    @PrimaryGeneratedColumn()
    id: number;

    @Column({unique: true, nullable: true})
    email?: string;

    @Column({name: "first_name"})
    firstName: string;

    @Column({name: "last_name"})
    lastName: string;

    @Column("enum", {default: Gender.Other})
    gender: Gender

    @Column({name: "joined_on", default: () => "CURRENT_TIMESTAMP", type: "timestamp"})
    joinedOn: Date

    @Column({unique: true, nullable: true})
    access?: number

    @Column({unique: true, nullable: true})
    sid?: number

    @Column({default: true})
    newsletter: boolean

    @Column({name: "doing_it", default: false})
    doingIT: boolean

    @Column({default: true})
    registered: boolean

    @OneToMany(type => AttendanceEntity, att => att.member)
    eventsAttended?: AttendanceEntity[]

}